### ISSUE TITLE
Test suite crashes with spaces in path to Python binary.

### ISSUE THREAD

**Issue by will-ca** [cython/cython#3369]
```
runTest (__main__.EndToEndTest)
End-to-end fused_cmethods ... /home/user/Data/Projects/Code/Cython 20_2_18/Python2Venv/bin/python setup.py build_ext -i

/bin/sh: /home/user/Data/Projects/Code/Cython: No such file or directory


FAIL

======================================================================
FAIL: runTest (__main__.EndToEndTest)
End-to-end fused_cmethods
----------------------------------------------------------------------
Traceback (most recent call last):
  File "runtests.py", line 1799, in runTest
    self.assertEqual(0, res, "non-zero exit status")
AssertionError: non-zero exit status

----------------------------------------------------------------------
Ran 25 tests in 15.594s
```

Kind of specific and probably wouldn't come up when using system Python, but could conceivably happen with `virtualenv`/`venv`, portable Python installations, etc.

**Comment by McSinyx** [comment]
The fix can be made by spawning commands as a list of strings, let me see if I can work on that.

**Comment by scoder** [comment]
It's a bit tricky, because the command is a single line in the test file, i.e. one string.
However, the `python` command is inserted externally. We could just quote that.

**Comment by scoder** [comment]
Closed by #3372.
